### ISSUE TITLE
"all: true" ignores include flag

### ISSUE THREAD

**Issue by jwalton** [istanbuljs/nyc#782]
## Expected Behavior

With this config file:

```
{
    "require": ["babel-register"],
    "reporter": ["lcov", "text-summary"],
    "extension": [".jsx", "*.js"],
    "include": ["src/**"],
    "all": true,
    "sourceMap": false,
    "instrument": false
}
```

I would expect only files in the 'src' folder to be instrumented.

## Observed Behavior

In my project, I see nyc run out of memory because it tries to instrument everything.  If I add a `console.log()` to [this line](https://github.com/istanbuljs/nyc/blob/18d559994afa5ab1be1a7d79d4543b13f1eec08b/index.js#L248) I see that webpack built files in /dist are being instrumented.

## Forensic Information

**Operating System:** OS/X
**Environment Information:**
node: 8.9.4
nyc: 11.4.1
output: https://gist.github.com/jwalton/193c1fcd02ad861d8805047e193b93f0

**Comment by JSteitz** [comment]
Have the same problem

As a workaround you can do the following:

**1. Exclude all directories except source**
```json
{
    "all": true,
    "exclude": [
        "test",
        "build",
        "coverage",
        "dist",
        etc..
    ]
}
```

**2. Change the root path** *(Currently I'm using this method)*
```json
{
    "all": true,
    "cwd": "./source"
  }
```

**Comment by bcoe** [comment]
@jwalton @JSteitz this definitely feels like a bug to me. Will happily accept a patch if either of you have a chance to dig into the issue before me.

**Comment by peterjgrainger** [comment]
I'm happy to pick this up.

Would the change be here: https://github.com/istanbuljs/nyc/blob/master/index.js#L179

The change would be to use the `includes` list if defined.  If it isn't then use `this.cmd`?

**Comment by coreyfarrell** [comment]
I believe this will be fixed by #914?
